python/ray/serve/backend_state.py

```diff
@@ -598,15 +600,14 @@ def __init__(self, controller_name: str, detached: bool,
         self._replica_constructor_retry_counter: Dict[BackendTag,
                                                       int] = defaultdict(int)
 
-        checkpoint = self._kv_store.get(CHECKPOINT_KEY)
-        if checkpoint is not None:
-            (self._replicas, self._backend_metadata,
-             self._backend_matadata_backup, self._deleted_backend_metadata,
-             self._target_replicas, self._target_versions,
-             self._backend_goals) = cloudpickle.loads(checkpoint)
+        self._resume_from_checkpoint()
```

Comment:
Should this be in an `else` condition? We don't want both recovery logics running at once right?

Reply:
We also still want to be able to recover the backend_state for the case where we use actor names

Reply:
i initially had if else there too but then realized we have to get some info from checkpoint anyways, like target replica or replica version. Only _replicas is invalid that we need to override with actor names

Reply:
Ah yeah I think we should split the checkpoint into two: the "target state" and the "current state" where current state is the one we're trying to get rid of